dev-support/code-formatter/hadoop_idea_formatter.xml

```diff
@@ -0,0 +1,77 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+-->
+<code_scheme name="Hadoop" version="173">
+  <option name="RIGHT_MARGIN" value="100" />
+  <JavaCodeStyleSettings>
+    <option name="CLASS_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
+    <option name="NAMES_COUNT_TO_USE_IMPORT_ON_DEMAND" value="999" />
+    <option name="PACKAGES_TO_USE_IMPORT_ON_DEMAND">
+      <value />
+    </option>
+    <option name="IMPORT_LAYOUT_TABLE">
+      <value>
+        <package name="java" withSubpackages="true" static="false" />
+        <package name="javax" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="org.apache.hadoop.thirdparty" withSubpackages="true" static="false" />
+        <package name="" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="org.apache" withSubpackages="true" static="false" />
+        <emptyLine />
+        <package name="" withSubpackages="true" static="true" />
+      </value>
+    </option>
+  </JavaCodeStyleSettings>
+  <codeStyleSettings language="JAVA">
+    <option name="RIGHT_MARGIN" value="100" />
+    <option name="KEEP_LINE_BREAKS" value="false" />
+    <option name="KEEP_FIRST_COLUMN_COMMENT" value="false" />
+    <option name="KEEP_CONTROL_STATEMENT_IN_ONE_LINE" value="false" />
+    <option name="KEEP_BLANK_LINES_IN_DECLARATIONS" value="1" />
+    <option name="KEEP_BLANK_LINES_IN_CODE" value="1" />
+    <option name="KEEP_BLANK_LINES_BEFORE_RBRACE" value="1" />
+    <option name="INDENT_CASE_FROM_SWITCH" value="false" />
+    <option name="ALIGN_MULTILINE_PARAMETERS" value="false" />
+    <option name="ALIGN_MULTILINE_RESOURCES" value="false" />
+    <option name="SPACE_WITHIN_ARRAY_INITIALIZER_BRACES" value="true" />
+    <option name="SPACE_BEFORE_ARRAY_INITIALIZER_LBRACE" value="true" />
+    <option name="CALL_PARAMETERS_WRAP" value="1" />
+    <option name="METHOD_PARAMETERS_WRAP" value="1" />
+    <option name="RESOURCE_LIST_WRAP" value="5" />
+    <option name="EXTENDS_LIST_WRAP" value="1" />
+    <option name="THROWS_LIST_WRAP" value="1" />
+    <option name="EXTENDS_KEYWORD_WRAP" value="1" />
+    <option name="THROWS_KEYWORD_WRAP" value="1" />
+    <option name="METHOD_CALL_CHAIN_WRAP" value="1" />
+    <option name="BINARY_OPERATION_WRAP" value="1" />
+    <option name="BINARY_OPERATION_SIGN_ON_NEXT_LINE" value="true" />
+    <option name="TERNARY_OPERATION_WRAP" value="5" />
+    <option name="ARRAY_INITIALIZER_WRAP" value="1" />
+    <option name="ASSIGNMENT_WRAP" value="1" />
+    <option name="METHOD_ANNOTATION_WRAP" value="1" />
+    <option name="CLASS_ANNOTATION_WRAP" value="1" />
```

Comment:
Is this required?
Will it not pull the annotation on the same line
```
  @Before public void setup() throws IOException {
```
We keep it like:
```
  @Before
  public void setup() throws IOException {
    ecPolicy = getEcPolicy();
```

Reply:
Good point, yes this is required only to keep it like as you mentioned.
Basically, default value for option `CLASS_ANNOTATION_WRAP` is `0`, which keeps the annotation in line (as per your first example), and providing value `1` keeps the annotation in the line above class/method names (as per your second example, which is the preference).

Reply:
Hmm, For me the present code isn't behaving the expected way, on removing these two line, it does behave the expected way.
I couldn't find a genuine source to find out what is the default. But found a page:
According to this the default itself is ``WRAP_ALWAYS``
https://github.com/joewalnes/idea-community/blob/master/platform/lang-api/src/com/intellij/psi/codeStyle/CommonCodeStyleSettings.java#L691

And 1 seems to denote, WRAP_AS_NEEDED:
https://github.com/joewalnes/idea-community/blob/master/platform/lang-api/src/com/intellij/psi/codeStyle/CommonCodeStyleSettings.java#L623

May be I have messed up something. Will double check once.

Reply:
You are correct, it seems I messed up while testing this. The default is WRAP_ALWAYS, and for explicitly providing the value, I tried `2` and it worked and provided value as `WRAP_ALWAYS`. When I used `1`, I guess it was for an old profile and hence it didn't work, whereas while importing new profile itself, `1` does not convert to `WRAP_ALWAYS` (your above references are valid, thanks for checking this one explicitly).

Reply:
Since `VARIABLE_ANNOTATION_WRAP` has different default value, keeping all four options with values `2` (WRAP_ALWAYS). Although method, class and field have the same default value, explicitly keeping `2` for all four of them kind of keep them in the same group, for better readability.

Reply:
@ayushtkn are you fine with the latest commit? If so, let me remove eclipse formatter xml now, I think it's going to be difficult to get review on eclipse formatter. If anyone in the community insists, we can revisit it later as sub-task. WDYT?

Reply:
Makes sense to me